tests/test_manifest.py

```diff
@@ -2619,6 +2621,110 @@ def check(expected, group_filter, extra_filter=None):
     check((False, False, True), 'group-filter: [-ga]',
           extra_filter=['-gb'])
 
+#########################################
+# Manifest group-filter inheritance tests
+#
+# In schema version 0.9, "manifest: group-filter:" values -- and
+# therefore Manifest.group_filter values --are *NOT* inherited
+# across manifest imports. Only the top level manifest
+# group-filter has any effect.
+#
+# Shortly after the release, we ran into use cases that made it clear
+# this was a mistake.
+#
+# This behavior means that people who import a manifest with projects
+# that are inactive by default need to copy/paste the group-filter
+# value if they want that same default. That kind of leaky filter is
+# of no use to people who want to build on top of the default projects
+# list without knowing the details, especially across multiple
+# versions when the set of defaults may change.
+#
+# Schema version 0.10 will reverse that behavior: manifests which
+# request schema version 0.10 will get Manifest.group_filter
+# values that *ARE* inherited from imported manifests, by
+# appending these values in import order.
+#
+# For compatibility, manifests which explicitly request a 0.9 schema
+# version will get the old behavior. However, we'll also release a
+# west 0.9.1 which will warn about a missing schema-version in the top
```

Comment:
Based on this observation: https://github.com/zephyrproject-rtos/zephyr/pull/32106#discussion_r576206945
i'm not sure we can actually cut a `0.9.1` now.